### PR TITLE
Subclassed DeprecationWarning in exceptions.py

### LINKED ISSUE #2945: Improve our warnings in line with PEP 565
See astropy/astropy#8465

### PR DESCRIPTION
**Pull request by cquark7**
### Description

Fixes #2945

I subclassed `DeprecationWarning` in custom warning class `SunpyDeprecationWarning`.
This was the only change required to close this issue.

### CODE REVIEW

**Comment by Cadair** [comment]
I think we should only do this on py3.7+ though?

**Comment by nabobalis** [comment]
This doesn't change anything <3.6 tho does it? Maybe I am mis-understanding the PEP change.

**Comment by cquark7** [comment]
@Cadair I'm unsure about what would be best for sunpy ecosystem, but this can be made `PEP 565` compliant in this way:

```python3
DeprecationClass = DeprecationWarning if sys.version_info >= (3, 7) else FutureWarning


class SunpyDeprecationWarning(DeprecationClass, SunpyWarning):
    """
    A warning class to indicate a deprecated feature.
    """
```

Kindly share your opinion on this.

**Comment by Cadair** [comment]
@cquark7 yep, I think that approach is good.

File: sunpy/util/exceptions.py [hunk 1]
```diff
@@ -19,7 +21,11 @@ class SunpyUserWarning(UserWarning, SunpyWarning):
     """
 
 
-class SunpyDeprecationWarning(SunpyWarning):
+# for pep 565 compliance
```
Comment (nabobalis) [inline sunpy/util/exceptions.py]:
```suggestion
# For PEP 565 (https://www.python.org/dev/peps/pep-0565/) compliance.
```

**Comment by nabobalis** [comment]
Thanks @cquark7

**Pull request merged**